ui/src/pages/search/authors/index.tsx

```diff
@@ -0,0 +1,315 @@
+        error = message;
+    }
+
+    const swrKey = `/users?search=${encodeURIComponent((Array.isArray(query) ? query[0] : query) || '')}&limit=${
+        limit || '10'
+    }&offset=${offset || '0'}`;
+
+    return {
+        props: {
+            searchType,
+            query,
+            limit,
+            offset,
+            fallback: {
+                [swrKey]: results
+            },
+            error
+        }
+    };
+};
+
+type Props = {
+    searchType?: Types.SearchType;
+    query: string | null;
+    limit: string | null;
+    offset: string | null;
+    error: string | null;
+};
+
+const Search: Types.NextPage<Props> = (props): React.ReactElement => {
+    const router = Router.useRouter();
+    const searchInputRef = React.useRef<HTMLInputElement>(null);
+    // params
+    const [searchType] = React.useState(props.searchType);
+    const [query, setQuery] = React.useState(props.query ? props.query : '');
+    // param for pagination
+    const [limit, setLimit] = React.useState(props.limit ? parseInt(props.limit, 10) : 10);
+    const [offset, setOffset] = React.useState(props.offset ? parseInt(props.offset, 10) : 0);
+
+    // ugly complex swr key
+
+    const swrKey = `/users?search=${encodeURIComponent(query || '')}&limit=${limit || '10'}&offset=${offset || '0'}`;
+
+    const { data: { data: results = [] } = {}, error, isValidating } = useSWR(swrKey);
+
+    const handlerSearchFormSubmit: React.ReactEventHandler<HTMLFormElement> = async (e) => {
+        e.preventDefault();
+        const searchTerm = searchInputRef.current?.value || '';
+        setQuery(searchTerm);
+    };
+
+    const resetFilters = (e: React.MouseEvent<HTMLButtonElement>) => {
+        setQuery('');
+        searchInputRef.current && (searchInputRef.current.value = '');
+        setOffset(0);
+        setLimit(10);
+    };
+
+    React.useEffect(() => {
+        setOffset(0);
+    }, [query, limit]);
+
+    return (
+        <>
+            <Head>
+                <meta name="description" content={Config.urls.search.description} />
+                <meta name="keywords" content={Config.urls.search.keywords.join(', ')} />
+                <link rel="canonical" href={Config.urls.search.canonical} />
+                <title>{Config.urls.search.title}</title>
+            </Head>
+
+            <Layouts.Standard>
+                <section className="container mx-auto px-8 py-8 lg:gap-4 lg:pt-16 lg:pb-0">
+                    <Components.PageTitle text={`Search results ${query ? `for ${query}` : ''}`} />
+                </section>
+                <section
+                    id="content"
+                    className="container mx-auto grid grid-cols-1 px-8 lg:grid-cols-12 lg:gap-x-16 lg:gap-y-8"
+                >
+                    <fieldset className="col-span-12 mb-8 grid w-full grid-cols-12 items-end gap-y-4 gap-x-6 lg:mb-0 lg:gap-x-6 2xl:gap-x-10">
+                        <legend className="sr-only">Search options</legend>
+
```

Comment:
We should extract this search markup and logic in a new component and use that component in both: /publications and /authors